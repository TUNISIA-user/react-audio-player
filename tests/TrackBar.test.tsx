import React from "react";
import { expect, test } from "vitest";
import renderer from "react-test-renderer"
import TrackBar from "../src/components/TrackBar";

describe("Test trackbar", () => {
  test("Trackbar snapshot", () => {
    const trackbar = renderer.create(
      <TrackBar 
        total={30}
        current={0}
        setCurrent={(num) => console.log(num)}
        showTrack={true}
        showKnob={true}
        trackHeight={1}
      />
    )
    const tree = trackbar.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
