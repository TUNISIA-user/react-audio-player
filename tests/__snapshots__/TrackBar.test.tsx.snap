// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test trackbar > Trackbar snapshot 1`] = `
<div
  data-testid="trackbar"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "gridColumn": 1,
      "gridRow": 1,
      "height": "15px",
      "placeSelf": "center",
      "width": "100%",
    }
  }
  tabIndex={0}
>
  <div
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(140, 140, 140)",
        "cursor": "pointer",
        "display": "flex",
        "height": 1,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        {
          "backgroundColor": "rgba(140, 140, 140)",
          "borderRadius": "50%",
          "cursor": "pointer",
          "height": "10px",
          "left": "0%",
          "marginLeft": -5,
          "opacity": 1,
          "position": "absolute",
          "transition": "opacity 0.75s ease-in-out",
          "width": "10px",
        }
      }
    />
  </div>
</div>
`;
